packages/ckeditor5-engine/tests/model/utils/insertcontent.js

```diff
@@ -1299,128 +1302,1629 @@ describe( 'DataController utils', () => {
+
+				expect( getData( model ) ).to.equal( '[<imageBlock></imageBlock>]<imageBlock></imageBlock>' );
+
+				expect( expectedMarker.getRange().start.path ).to.deep.equal( [ 1 ] );
+				expect( expectedMarker.getRange().end.path ).to.deep.equal( [ 2 ] );
+			} );
+
+			it( 'should create marker on imageBlock after inserting into paragraph next to another imageBlock', () => {
+				// <imageBlock></imageBlock><paragraph>[]</paragraph>
+				// {<imageBlock></imageBlock>}
+				//
+				// <imageBlock></imageBlock>{<imageBlock></imageBlock>}
+				setData( model, '<imageBlock></imageBlock><paragraph>[]</paragraph>' );
+
+				insertHelper( '<imageBlock></imageBlock>', {
+					'marker-a': { start: [ 0 ], end: [ 1 ] }
+				} );
+
+				const expectedMarker = model.markers.get( 'marker-a' );
+
+				expect( getData( model ) ).to.equal( '<imageBlock></imageBlock>[<imageBlock></imageBlock>]' );
+
+				expect( expectedMarker.getRange().start.path ).to.deep.equal( [ 1 ] );
+				expect( expectedMarker.getRange().end.path ).to.deep.equal( [ 2 ] );
+			} );
+
+			it( 'should create marker on imageBlock after inserting between another image blocks', () => {
+				// <imageBlock></imageBlock>[]<imageBlock></imageBlock>
+				// {<imageBlock></imageBlock>}
+				//
+				// <imageBlock></imageBlock>{<imageBlock></imageBlock>}<imageBlock></imageBlock>
+				setData( model, '<imageBlock></imageBlock>[]<imageBlock></imageBlock>' );
+
+				insertHelper( '<imageBlock></imageBlock>', {
+					'marker-a': { start: [ 0 ], end: [ 1 ] }
+				}, [ 1 ] );
+
+				const expectedMarker = model.markers.get( 'marker-a' );
+
+				expect( getData( model ) ).to.equal( '[<imageBlock></imageBlock>]<imageBlock></imageBlock><imageBlock></imageBlock>' );
+
+				expect( expectedMarker.getRange().start.path ).to.deep.equal( [ 1 ] );
+				expect( expectedMarker.getRange().end.path ).to.deep.equal( [ 2 ] );
+			} );
+
+			it( 'should create marker on imageBlock after inserting next to imageInline', () => {
+				// <paragraph><imageInline></imageInline>[]</paragraph>
+				// {<imageBlock></imageBlock>}
+				//
+				// <paragraph><imageInline></imageInline></paragraph>{<imageBlock></imageBlock>}
+				setData( model, '<paragraph><imageInline></imageInline>[]</paragraph>' );
+
+				insertHelper( '<imageBlock></imageBlock>', {
+					'marker-a': { start: [ 0 ], end: [ 1 ] }
+				} );
+
+				const expectedMarker = model.markers.get( 'marker-a' );
+
+				expect( getData( model ) ).to.equal( '<paragraph><imageInline></imageInline></paragraph>[<imageBlock></imageBlock>]' );
+
+				expect( expectedMarker.getRange().start.path ).to.deep.equal( [ 1 ] );
+				expect( expectedMarker.getRange().end.path ).to.deep.equal( [ 2 ] );
+			} );
+
+			it( 'should create marker on imageBlock and imageInline after inserting', () => {
+				// <paragraph>[]</paragraph>
+				// {<imageBlock></imageBlock><imageInline></imageInline>}
+				//
+				// {<imageBlock></imageBlock><paragraph><imageInline></imageInline></paragraph>}
```

Comment:
This is an interesting, although rather not probable case. Interesting as it is one big marker and the inline image will need to be paragraphed, so the marker has to be on `<paragraph>` but it does not have to be on `</paragraph>`.

I think this will get also fixed when we will change from basing on `affectedRange` to some more precise solution.

Reply:
For the moment, I don't care much about this scenario as I think it will change. You can leave it like this (and it should start failing after we improve our solution).